.github/workflows/rc_sync.yml

```diff
@@ -0,0 +1,73 @@
+# This workflow will sync changes in the release candidate branch to master during a
+# feature freeze. The action is run daily and will only create a pull request when
+# there exists a release candidate branch corresponding to the current package version
+# as declared in pennylane/_version.py, that is if the branch "v0.23.0-rc0" exists while
+# the package version is "v0.23.0-dev".
+
+name: Sync rc with master
+
+# Controls when the workflow will run
+on:
+  # Scheduled trigger every weekday at 2:47am UTC
+  schedule:
+  - cron:  '47 2 * * 1-5'
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job to sync the master branch with changes from the rc
+  sync:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out the PennyLane repository
+      - uses: actions/checkout@v3
+
+      # Check for the existence of an rc branch for the current PennyLane version
+      - name: Check for rc branch
+        run: |
+          VERSION=$(python setup.py --version)
+          IFS=. read MAJ MIN PAT <<< "${VERSION%.dev0}"
+          RC_BRANCH="v${MAJ}.$((MIN-1)).${PAT}-rc0"
+          if git ls-remote --exit-code origin "refs/heads/$RC_BRANCH"; then
+            echo "branch_exists=true" >> $GITHUB_ENV
+            echo "rc_branch=$RC_BRANCH" >> $GITHUB_ENV
+          else
+            echo "branch_exists=false" >> $GITHUB_ENV
+          fi
+
+      # Create a new branch at from the current rc to be used in the PR
+      - name: Create temp branch
+        if: ${{ env.branch_exists == 'true' }}
+        run: |
+          git fetch
+          git checkout "${{ env.rc_branch }}"
+          BRANCH="rc_$(date +'%Y-%m-%d-%H-%M-%S')"
+          echo "tmp_branch=$BRANCH" >> $GITHUB_ENV
+          git checkout -b "$BRANCH"
+          git checkout master -- pennylane/_version.py
+          git checkout master -- doc/releases/changelog-dev.md
```

Comment:
Worth leaving a comment here on why we're doing this. :+1:

Reply:
Sounds good 👍